test/fluree/db/query/sql_test.cljc

```diff
@@ -150,7 +150,35 @@
                     ["?person" "person/name" "?personName"]
                     ["?person" "person/email" "?personEmail"]]
                    (:where subject))
-                "correctly constructs the where clause")))))
+                "correctly constructs the where clause"))))
+
+      (testing "with aggregate functions"
+        (testing "count"
+          (testing "explicit fields"
+            (let [query   "SELECT COUNT(middleName) FROM person"
+                  subject (parse query)]
+
+              (is (= ["(count ?personMiddleName)"]
+                     (:select subject))
+                  "correctly constructs the select clause")
+
+              (is (= [["?person" "rdf:type" "person"]
+                      ["?person" "person/middleName" "?personMiddleName"]]
+                     (:where subject))
+                  "correctly constructs the where clause"))
+
+            (testing "with distinct"
+              (let [query   "SELECT COUNT(DISTINCT middleName) FROM person"
+                    subject (parse query)]
+
+                (is (= ["(count (distinct ?personMiddleName))"]
+                     (:select subject))
+                  "correctly constructs the select clause")
+
+              (is (= [["?person" "rdf:type" "person"]
+                      ["?person" "person/middleName" "?personMiddleName"]]
+                     (:where subject))
+                  "correctly constructs the where clause")))))))
```

Comment:
Would it be useful to add a test for `COUNT(*)` as well?

Reply:
Actually, the parser doesn't yet support wildcards with set functions (e.g. `COUNT(*)`) yet. That's another branch in the [parse tree](https://github.com/fluree/db/blob/feature/sql-aggregates/resources/sql-92.bnf#L62-L64).

This patch [implements the `general-set-function` branch](https://github.com/fluree/db/blob/feature/sql-aggregates/src/fluree/db/query/sql.cljc#L199-L218), but I haven't gotten to the other branch that represents a `COUNT(*)` parse yet.